MetaMask/metamask-extension
# Adding price checker deactivation for Optimism

## Explanation
Fixes #18562 

Adding the `usecurrencyRate` check on `detailText` for layer1 fees and total on <MultilayerFeeMessage/> to block the fiat values.

## Manual Testing Steps

1. Go to Privacy Settings
2. Disable the Price checker
3. Close Settings
4. Try to Send a tx
5. Make sure that fiat conversion does not appear on confirmation screens
